### ISSUE TITLE
desi_tile_qa MINTFRAC failure

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1599]
```
source /global/cfs/cdirs/desi/software/desi_environment.sh 22.1a
module swap desispec/0.50.0
export DESI_SPECTRO_REDUX=/global/cfs/cdirs/desi/spectro/redux
export SPECPROD=f5

desi_tile_qa --group pernight -n 20210508 -t 80872 --outdir .
...
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20211217-2.0.0/code/desispec/0.50.0/bin/desi_tile_qa", line 267, in <module>
    main()
  File "/global/common/software/desi/cori/desiconda/20211217-2.0.0/code/desispec/0.50.0/bin/desi_tile_qa", line 196, in main
    entry = func(**func_arg)
  File "/global/common/software/desi/cori/desiconda/20211217-2.0.0/code/desispec/0.50.0/bin/desi_tile_qa", line 62, in func
    fiberqa_table , petalqa_table = compute_tile_qa(night,tileid,specprod_dir,exposure_qa_dir,group=group)
  File "/global/common/software/desi/cori/desiconda/20211217-2.0.0/code/desispec/0.50.0/lib/python3.9/site-packages/desispec/tile_qa.py", line 219, in compute_tile_qa
    minimal_efftime = qa_params["tile_qa"]["fiber_rel_mintfrac"]*exposure_qa_meta["MINTFRAC"]*exposure_qa_meta["GOALTIME"]
KeyError: 'MINTFRAC'
```

I'm guessing that this error applies to all tiles of that era, before MINTFRAC was added to fiberassign files.  @araichoor please check.

**Comment by araichoor** [comment]
Yes, sure... basically, lots of those tile_qa routines are irrelevant for such tiles.
What do we want to do here? 
I've given a try to a small fix:
- GOALTIME: if missing, set GOALTIME=1000/150/30 for dark,cmx/bright/backup, using the TARG header keyword from the fiberassign-TILEID.fits.gz file; if no fiberassign file or no TARG keyword or no correct TARG value, raise an error;
- MINTFRAC: if missing, set MINTFRAC=0.9.
Those are ~nominal values, see here: https://github.com/desihub/fiberassign/blob/c9ef922dae336860959ae8cf1d2c07b1b8eb99a3/bin/fba_launch#L40-L66

Would that make sense? That allows the `tile-qa-TILEID*fits` file to be generated.
But then the code returns an `ERROR`, as there is no `SURVEY` keyword in the header:

```
INFO:desi_tile_qa:67:func: wrote /global/cfs/cdirs/desi/users/raichoor/tmpdir/tiles/pernight/80872/20210508/tile-qa-80872-20210508.fits
INFO:tile_qa_plot.py:978:make_tile_qa_plot: no SURVEY keyword in header, skip this tile
ERROR:desi_tile_qa:113:_wrap_make_tile_qa_plot: failed to compute QA plot for /global/cfs/cdirs/desi/users/raichoor/tmpdir/tiles/pernight/80872/20210508/tile-qa-80872-20210508.fits
INFO:desi_tile_qa:219:main: Trying again on tile-qa-80872-20210508.png
INFO:tile_qa_plot.py:978:make_tile_qa_plot: no SURVEY keyword in header, skip this tile
ERROR:desi_tile_qa:113:_wrap_make_tile_qa_plot: failed to compute QA plot for /global/cfs/cdirs/desi/users/raichoor/tmpdir/tiles/pernight/80872/20210508/tile-qa-80872-20210508.fits
```

**Comment by sbailey** [comment]
Using defaults for GOALTIME and MINTFRAC if missing inputs for the purposes of tile-qa sounds fine.

Regarding SURVEY, see desispec.io.meta.faflavor2program .  Perhaps we need the equivalent of faflavor2survey?  In bin/desi_tsnr_afterburner has a chain of logic depending upon whether FAFLAVOR or FAPRGRM is available, with code like
```
        if entry["SURVEY"]=="unknown" :
            if faflavor.find("sv1")>=0. : entry["SURVEY"]="sv1"
            elif faflavor.find("sv2")>=0. : entry["SURVEY"]="sv2"
            elif faflavor.find("cmx")>=0. : entry["SURVEY"]="cmx"
...
    if entry['FAPRGRM'].startswith('dith'):
        if entry['SURVEY'] == 'unknown':
            entry['SURVEY'] = 'cmx'
```
At minimum we want something that doesn't crash desi_tile_qa, which could just be "if SURVEY is missing we don't make plots A, B, and C...", e.g. maybe we make positioning plots but not n(z).  i.e. ok to proceed with a minimal fix to avoid crashes even if a better fix might come later.

**Comment by araichoor** [comment]
OK, I think I've something working, with your latter suggestion, I'll open a PR.

**Comment by sbailey** [comment]
fixed by PR #1603
